utils/config.go

```diff
@@ -462,6 +458,30 @@ func (config *Config) evaluateCertificate() error {
 	return nil
 }
 
+// evaluateAPIEndpointURL evaluates if API endpoint url is valid, advising if invalid version defined, and adapting if required
+func (config *Config) evaluateAPIEndpointURL() error {
+	log.Debug("evaluateAPIEndpointURL")
+
+	// remove ending slash if exist
+	config.APIEndpoint = strings.TrimRight(config.APIEndpoint, "/")
+
+	// In User mode, endpoint url should include API version
+	if !config.IsAgentMode() {
+		cURL, err := url.Parse(config.APIEndpoint)
+		if err != nil {
+			return err
+		}
+		if cURL.Path == "" {
+			config.APIEndpoint = strings.Join([]string{config.APIEndpoint, VERSION_API_USER_MODE}, "/")
+			log.Warnf("Defined API server endpoint url does not include API version. Normalized to latest version (%s): %s", VERSION_API_USER_MODE, config.APIEndpoint)
+		} else if cURL.Path != strings.Join([]string{"/", VERSION_API_USER_MODE}, "") {
+			log.Warnf("Defined API server endpoint url does not match the latest API version (%s). Received %s", VERSION_API_USER_MODE, cURL.Path)
```

Comment:
Rather than received, I'd type Found. Also, I'm not sure we should be talking about latest API version, but about the latest API version the tool is aware of: when we release a v3, this commit's code would show a warning about v3 not matching latest version v2 if the user set it up to use v3.

Reply:
Done, updated in order to clarify message